workers/data_refinery_workers/processors/gene_convert_illumina.R

```diff
@@ -0,0 +1,71 @@
+###
+# gene_convert_illumina.R
+# Alex's Lemonad Stand Foundation
+# Childhood Cancer Data Lab
+#
+###
+
+# import magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+#######################
+# The command interface!
+#######################
+
+library("optparse")
+library(data.table)
+library("dplyr")
+library("rlang")
+suppressPackageStartupMessages(library(AnnotationDbi))
+
+option_list = list(
+  make_option(c("-p", "--platform"), type="character", default="", 
+              help="Platform", metavar="character"),
+  make_option(c("-i", "--inputFile"), type="character", default="", 
+              help="inputFile", metavar="character"),
+  make_option(c("-o", "--outputFile"), type="character", default="", 
+              help="outputFile", metavar="character")
+)
+
+opt_parser = OptionParser(option_list=option_list);
+opt = parse_args(opt_parser);
+
+platform <- opt$platform
+filePath <- opt$inputFile
+outFilePath <- opt$outputFile
+
+# Read the data file
+message("Reading data file...")
+suppressWarnings(exprs <- fread(filePath, 
+					stringsAsFactors=FALSE, 
+					sep="\t", header=TRUE, 
+					autostart=10, 
+					data.table=FALSE, 
+					check.names=FALSE, 
+					fill=TRUE, 
+					na.strings="", 
+					showProgress=FALSE)
+				)
+
+message("Here's the data..")
+message(head(exprs))
+
+# Build a PROBEID:ENSEMBL map
+message("Here's the db..")
+db_name <- paste(platform, ".db", sep="")
+message(db_name)
+library(db_name, character.only=TRUE)
+
+# Replace the probe IDs with Ensembles
+exprs[, 1] <- mapIds(get(db_name), keys=exprs[, 1], column="ENSEMBL", keytype="PROBEID")# , multiVals="first")
```

Comment:
If we use [`mapIds` with the default behavior (we don't set `multiVals` argument)](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.34.4/topics/AnnotationDb-objects) and there are multiple matches (so here it would be a single probe maps to multiple Ensembl gene identifiers), it selects the whatever comes back first. I'm not sure how frequent one-to-multiple mappings are, have you come across that?

Reply:
It doesn't yell about one-to-multiple mappings are far as I can see, but I'm not sure if that error would make it to the logs.

What mode for multiVals would you prefer?

Reply:
What if we did something like (relies on tidyverse packages):
```R
# need what the first column is called
exprs_id_name <- colnames(exprs)[1]
# don't replace the identifiers in exprs yet
mapped_list <- mapIds(get(db_name), keys=exprs[, 1], column="ENSEMBL", keytype="PROBEID", multiVals="list")
# get into data.frame form, should capture one to multiple mapping
mapped_df <- reshape2::melt(mapped_list)
# not 100% sure this bit is correct -- might be the other way around
colnames(mapped_df) <- c("ENSEMBL", "PROBEID")
annot_exprs_df <- mapped_df %>%
  dplyr::filter(!is.na(ENSEMBL)) %>%   # dropping unmapped -- mapIds might do this by default!
  dplyr::inner_join(y = exprs, by = c("PROBEID" = exprs_id_name)) %>%  # annotation heavy lifting
  dplyr::select(-PROBEID)  # drop the probe ids
```
And then you'd write `annot_exprs_df` to file. I haven't fully tested this out, but I think it's close.